src/dfilemaker/dfilemaker.c

```diff
@@ -580,25 +653,106 @@ int main(int narg, char** arg)
     MPI_Datatype dirname_type;
     MPI_Datatype tname_type;
 
-    /*----------------------------------------------
-     * get nfiles = number of files to create basic
-     *----------------------------------------------*/
-    if (narg < 4) {
-        if (rank == 0) {
-            printf("Usage: dfilemaker <nitems> <nlevels> <maxflen> [-i seed]\n");
-        }
-        MPI_Finalize();
-        exit(0);
-    }
-    ntotal  = atoi(arg[1]);
-    nlevels = atoi(arg[2]);
-    maxflen = atoi(arg[3]);
-    c=getopt(narg,arg,"i:");
-    if (narg > 1)
+   /*----------------------------------------------
+    * get nfiles = number of files to create basic
+    *----------------------------------------------*/
+
+   /*---------------------
+    *  loop over options  
+    *---------------------*/
+    while (1)
     {
-       if (c=='i') jseed = atoi(optarg);
-       if (jseed) iseed=jseed;
-    }
+       /*---------------------------------------------------------------------
+        *  shortopts below are followed by a colon if they take an argument  
+        *---------------------------------------------------------------------*/        
+        c=getopt_long(narg,arg,"i:f:d:n:r:s:w:vh",long_options,&longind);
+        if (c <= 0) break;
+        minterm=(char*)MFU_MALLOC(10*sizeof(char));
+        maxterm=(char*)MFU_MALLOC(10*sizeof(char));
+        switch (c) {
+           case 'i':
+             jseed = atoi(optarg);
+             if (jseed) iseed=jseed;
+             break;
+           case 'f':
+             for (kft=0;kft<4;kft++) if (strcmp(optarg,filltype[kft])==0) break;
```

Comment:
To match style we're using in the code base, let's add braces to all for/if/while statements and place the code block indented on the following lines.